File: 66_plus_one/plusOne.go
package plusone

func plusOne(digits []int) []int {
	for i := len(digits) - 1; i >= 0; i-- {
		if digits[i]+1 < 10 {
			digits[i] += 1
			return digits
		}

		digits[i] = 0

		if i == 0 {
			res := []int{1}
			res = append(res, digits...)
			return res
		}
	}

	return digits
}

File: 66_plus_one/plus_one.rs
impl Solution {
    pub fn plus_one(mut digits: Vec<i32>) -> Vec<i32> {
        for x in digits.iter_mut().rev() {
            if *x + 1 < 10 {
                *x += 1;
                return digits;
            }

            *x =  0;
        }

        digits.insert(0, 1);
        digits
    }
}